test/lib/modules/endpointRunner.test.js

```diff
@@ -0,0 +1,367 @@
+        .matchHeader('x-kinvey-api-version', '3')
+        .post(`/rpc/${environmentId}/custom/myEndpoint`)
+        .basicAuth({
+          user: environmentId,
+          pass: mastersecret
+        })
+        .reply(200, [{ _id: 123, someData: 'abc' }, { _id: 456, someData: 'xyz' }]);
+
+      const endpoint = this.runner().endpoint('myEndpoint');
+      (endpoint.execute({})).should.be.a.Promise(); // eslint-disable-line new-cap
+      return done();
+    });
+
+    it('should execute an endpoint', (done) => {
+      nock('https://baas.kinvey.com')
+        .matchHeader('content-type', 'application/json')
+        .matchHeader('x-kinvey-api-version', '3')
+        .post(`/rpc/${environmentId}/custom/myEndpoint`)
+        .basicAuth({
+          user: environmentId,
+          pass: mastersecret
+        })
+        .reply(200, [{ _id: 123, someData: 'abc' }, { _id: 456, someData: 'xyz' }]);
+
+      const endpoint = this.runner().endpoint('myEndpoint');
+      endpoint.execute({}, (err, result) => {
+        should.not.exist(err);
+        result.should.containDeep([{ _id: 123, someData: 'abc' }, { _id: 456, someData: 'xyz' }]);
+        return done();
+      });
+    });
+
+    it('should include a body', (done) => {
+      nock('https://baas.kinvey.com')
+        .matchHeader('content-type', 'application/json')
+        .matchHeader('x-kinvey-api-version', '3')
+        .post(`/rpc/${environmentId}/custom/myEndpoint`, { foo: 'bar' })
+        .basicAuth({
+          user: environmentId,
+          pass: mastersecret
+        })
+        .reply(200, [{ _id: 123, someData: 'abc' }, { _id: 456, someData: 'xyz' }]);
+
+      const endpoint = this.runner().endpoint('myEndpoint');
+      endpoint.execute({ foo: 'bar' }, (err, result) => {
+        should.not.exist(err);
+        result.should.containDeep([{ _id: 123, someData: 'abc' }, { _id: 456, someData: 'xyz' }]);
+        return done();
+      });
+    });
+
+    it('should include empty object as body if null body passed', (done) => {
+      nock('https://baas.kinvey.com')
+        .matchHeader('content-type', 'application/json')
+        .matchHeader('x-kinvey-api-version', '3')
+        .post(`/rpc/${environmentId}/custom/myEndpoint`, {})
+        .basicAuth({
+          user: environmentId,
+          pass: mastersecret
+        })
+        .reply(200, [{ _id: 123, someData: 'abc' }, { _id: 456, someData: 'xyz' }]);
+
+      const endpoint = this.runner().endpoint('myEndpoint');
+      endpoint.execute(null, (err, result) => {
+        should.not.exist(err);
+        result.should.containDeep([{ _id: 123, someData: 'abc' }, { _id: 456, someData: 'xyz' }]);
+        return done();
+      });
+    });
+
+    it('should include empty object as body if no body passed', (done) => {
+      nock('https://baas.kinvey.com')
+        .matchHeader('content-type', 'application/json')
+        .matchHeader('x-kinvey-api-version', '3')
+        .post(`/rpc/${environmentId}/custom/myEndpoint`, {})
```

Comment:
No request body validation?

Reply:
This test tests if the body is empty, not sure what you're asking...